sktime/forecasting/multiplexer.py

```diff
@@ -0,0 +1,173 @@
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexerForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.component_estimators = component_estimators
+        self.selected_estimator = selected_estimator
+        self._check_components()
+        self._forecaster_fit_params = None
+
+        super(MultiplexerForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.component_estimators, list):
+            raise Exception(
+                "Please provide a list " "for component_estimators composed of tuples"
+            )
+
+        for component in self.component_estimators:
+            name, estimator = component
+            if not isinstance(estimator, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(estimator, name)
+                )
+
+    def _check_fit_params(self, fit_params):
+        if fit_params is None or fit_params == {}:
+            return
+
+        for component in self.component_estimators:
+            name, _ = component
+
+            if name not in fit_params.keys():
+                raise KeyError(
+                    "If you provide fit_params for models "
+                    "dictionary key of fit params need to "
+                    " match the associated component key."
+                )
+
+    def _check_selected_estimator(self):
+        component_names = [name for name, _ in self.component_estimators]
+        if self.selected_estimator not in component_names:
+            raise Exception(
+                "Please check the selected_estimator argument provided "
+                " Valid selected_estimator parameters: {}".format(component_names)
+            )
+
+    def _update_forecaster_fit_params(self, fit_params):
+        self._check_fit_params(fit_params)
+
+        if self.selected_estimator is None or fit_params is None:
+            self._forecaster_fit_params = None
+            return
+
+        if self.selected_estimator in fit_params.keys():
+            self._forecaster_fit_params = self.component_fit_params[
+                self.selected_estimator
+            ]
+        else:
+            self._forecaster_fit_params = None
+
+    def _update_forecaster(self):
+        self._check_selected_estimator()
+        if self.selected_estimator is not None:
+            for name, estimator in self.component_estimators:
+                if self.selected_estimator == name:
+                    self._forecaster = copy.deepcopy(estimator)
```

Comment:
```suggestion
                    self._forecaster = clone(estimator)
```
where `clone` comes from `sklearn.base`